packages/web-react/src/components/FileUploader/useFileUploaderInput.ts

```diff
@@ -0,0 +1,162 @@
+import { ChangeEvent, DragEvent } from 'react';
+import { useDragAndDrop } from '../../hooks';
+import { useFileUploaderContext } from './FileUploaderContext';
+
+export interface UseFileUploaderInputProps {
+  accept?: string;
+  isMultiple?: boolean;
+  maxFileSize: number;
+  maxUploadedFiles: number;
+  onError?: (error: string) => void;
+}
+
+export interface UseFileUploaderInputReturn {
+  isDragging: boolean;
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  onDragEnter: (event: DragEvent<HTMLDivElement>) => void;
+  onDragLeave: (event: DragEvent<HTMLDivElement>) => void;
+  onDragOver: (event: DragEvent<HTMLDivElement>) => void;
+  onDrop: (event: DragEvent<HTMLDivElement>) => void;
+}
+
+export const useFileUploaderInput = (props: UseFileUploaderInputProps): UseFileUploaderInputReturn => {
+  const { maxFileSize, maxUploadedFiles, isMultiple, onError, accept } = props;
+
+  const { fileQueue, addToQueue, clearQueue, errorMessages } = useFileUploaderContext();
+
+  const getUpdatedFileName = (name: string): string => `attachment__${name.replace(/\./g, '_')}`;
+
+  const checkAllowedFileSize = (file: File) => {
+    if (file.size > maxFileSize) {
+      throw new Error(errorMessages?.errorMaxFileSize);
+    }
+  };
+
+  const checkFileDuplicity = (file: File) => {
+    if (isMultiple && fileQueue.has(getUpdatedFileName(file.name))) {
+      throw new Error(errorMessages?.errorFileDuplicity);
+    }
+  };
+
+  const checkQueueSize = () => {
+    if (fileQueue.size >= maxUploadedFiles) {
+      throw new Error(errorMessages?.errorMaxUploadedFiles);
+    }
+  };
+
+  const checkAllowedFileType = (file: File) => {
+    const fileExtension = file.name.split('.').pop()?.toLowerCase() as string;
+    const fileType = file.type;
+    let isSupported;
+    // Determine if validation should occur
+    // If the `accept` attribute is set and not all types are allowed
+    const shouldValidate = accept && !(accept === '' || accept === '*' || accept === '*/*');
+
+    if (!shouldValidate) {
+      isSupported = true;
+    } else {
+      // An array of all allowed file types and extensions
+      const acceptArray = accept.replace(/ /g, '').split(',');
+      // An array of all allowed file extensions
+      const acceptExtensions = acceptArray.filter((exp) => exp.match(/\./g));
+      // An array of all allowed file types
+      const acceptTypes = acceptArray.filter((exp) => exp.match(/\//g));
+
+      acceptExtensions.forEach((exp) => {
```

Comment:
Please, respect the logic we have in https://github.com/lmc-eu/spirit-design-system/pull/825/files#diff-f7cd05f847eba60d2b165b77e9fba736ca3995a7e5092e9b268bd34eb878fd6a

Reply:
Fixed